agent/provider/src/cli/keystore.rs

```diff
@@ -89,3 +98,53 @@ fn remove(config: ProviderConfig, remove: Remove) -> anyhow::Result<()> {
 fn cert_dir_path(config: &ProviderConfig) -> anyhow::Result<PathBuf> {
     Ok(config.cert_dir.get_or_create()?)
 }
+
+fn print_cert_list(json_output: bool, certs_data: Vec<util::CertBasicData>) {
+    let mut table_builder = CertTableBuilder::new();
+    for data in certs_data {
+        table_builder.with_row(data);
+    }
+    let table = table_builder.build();
+    let output = CommandOutput::from(table);
+    output.print(json_output);
+}
+
+struct CertTableBuilder {
+    columns: Vec<String>,
+    values: Vec<Value>,
+}
+
+impl CertTableBuilder {
+    pub fn new() -> Self {
+        let columns = vec![
+            "ID".to_string(),
+            "Not After".to_string(),
+            "Subject".to_string(),
+        ];
+        let values = vec![];
+        Self { columns, values }
+    }
+
+    pub fn with_row(&mut self, data: CertBasicData) {
+        let row = serde_json::json! {[ data.id, data.not_after, data.subject ]};
+        self.values.push(row);
+    }
+
+    pub fn build(self) -> ResponseTable {
+        let columns = self.columns;
+        let values = self.values;
+        ResponseTable { columns, values }
+    }
+}
+
+impl CertBasicDataVisitor for CertTableBuilder {
+    fn accept(&mut self, data: CertBasicData) {
+        self.with_row(data)
+    }
+}
+
+fn println_conditional(skip: bool, txt: &str) {
```

Comment:
I make same check 3 times, so to make it more difficult to accidentally break later and to make code a bit more readable I moved it to a function. It could take a whole `ProviderConfig`, but then this function would require more info than it actually needs (which is only a skip: bool field), which I assumed is worse. Will change it to take a whole `ProviderConfig`.